### PR TITLE
Discord bot

### PR DESCRIPTION
**Pull request by CatLuong0106**
HEY JOHN CHECK THIS PR

### CODE REVIEW

File: .devcontainer/.env [hunk 1]
```diff
@@ -7,6 +7,15 @@ S3_BUCKET_NAME=resumes
 S3_ROOT_ACCESS_KEY=minioadmin
 S3_ROOT_SECRET_KEY=minioadmin
 
+DATABASE_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}/${POSTGRES_DATABASE}?schema=public&connection_limit=10&pgbouncer=true
+SHADOW_DATABASE_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}/${POSTGRES_DATABASE}?schema=public
+
+DISCORD_TOKEN="MTE2NDY2NDkxMjI3MDg3MjY2Nw.Gb5Byz.LTMz54pcCIBvlUemNinMvN1rwCN85EQACqLcpg"
```
Comment (john-whiting) [inline .devcontainer/.env:13]: These discord environment variables should _not_ be committed. Please use the .env.local file instead for these (and put a reference to it in .env.local.default

Also, you will want to invalidate/regenerate this token in your Discord Developer Portal.

File: discord_bot/.env.example
Comment (john-whiting) [inline discord_bot/.env.example:1]: This file can be removed.

File: discord_bot/src/commands/ping.ts
Comment (john-whiting) [inline discord_bot/src/commands/ping.ts:1]: We can probably get rid of this ping command.

File: discord_bot/src/commands/schedule.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { Command } from "./index";
+
+export const scheduleCommand = {
+  data: new SlashCommandBuilder()
+    .setName("schedule")
+    .setDescription("Currently returns the value passed in")
+    .addIntegerOption(option => option.setName("value").setDescription("milliseconds to wait till scheduling")),
+  execute: async (interaction: ChatInputCommandInteraction) => {
+    const value = interaction.options.getInteger("value") ?? 1000;
+    const channel = await interaction.client.channels.fetch("895903170004418590") as TextChannel;
```
Comment (john-whiting) [inline discord_bot/src/commands/schedule.ts:11]: Should we be hardcoding this channel value?

File: discord_bot/src/commands/verify.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import { checkin } from '../config';
+
+
+
+const prisma = new PrismaClient();
+
+
+export const verifyCommand = {
+  data: new SlashCommandBuilder()
+    .setName("verify-new")
```
Comment (john-whiting) [inline discord_bot/src/commands/verify.ts:13]: Is this supposed to be "verify-new"?

File: discord_bot/src/commands/verify.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    .setDescription("Verify Users For MakeUC!")
+    .addStringOption(option =>
+      option.setName('email')
+        .setDescription('Insert the email address the hacker used to register for MakeUC')
+        .setRequired(true)),
+  execute: async (interaction: ChatInputCommandInteraction) => {
+    const email = interaction.options.getString('email');
+    console.log(email);
```
Comment (john-whiting) [inline discord_bot/src/commands/verify.ts:21]: Can we remove this console log?

File: discord_bot/src/commands/verify.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    let reply = "";
+
+    if (checkin === "close") {
```
Comment (john-whiting) [inline discord_bot/src/commands/verify.ts:24]: Can we change this to be `!== "open"`? Then, anything that isn't open will not count.

Comment (john-whiting) [reply discord_bot/src/commands/verify.ts:24]: We should also probably early return

File: discord_bot/src/commands/verify.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      reply = "Checkin is not open yet! Please wait until checkin starts. We appreciate your patience!";
+      await interaction.reply({ content: reply });
+    }
+    if (checkin === "open" && email) {
+      (async function () {
```
Comment (john-whiting) [inline discord_bot/src/commands/verify.ts:29]: Is there a reason you are using an anonymous function here?

File: discord_bot/src/commands/verify.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        try {
+          const participant = await prisma.registrant.findFirstOrThrow({
+            where: {
+              email: email,
+              registrationYear: 2023,
+            },
+          });
+
+          console.log(participant)
```
Comment (john-whiting) [inline discord_bot/src/commands/verify.ts:38]: Should we remove this console log?

File: discord_bot/src/commands/verify.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+          if (!participant) {
+            reply = `I could not find a registration with the email: ${email}. Please make sure that the email you entered is correct!`;
```
Comment (john-whiting) [inline discord_bot/src/commands/verify.ts:41]: What is the purpose of setting `reply` when only using it in one place? Also, should we be returning early here?

File: discord_bot/src/commands/verify.ts [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+            await interaction.reply({ content: reply });
+          } else if (participant.verified) {
+            reply = `Hello ${participant.firstName + participant.lastName}, you already checked in!`;
+            await interaction.reply({ content: reply });
+          } else {
+            reply = `Hello ${participant.firstName + participant.lastName}, you have not checked in yet. Please enter \`/checkin <email>\` in the verification channel!`;
+            await interaction.reply({ content: reply });
+            prisma.registrant
```
Comment (john-whiting) [inline discord_bot/src/commands/verify.ts:49]: What is this line doing?

File: discord_bot/src/config.ts [hunk 10]
```diff
@@ -0,0 +1,23 @@
+// import { config } from "dotenv";
+
+// config();
+
+import * as dotenv from 'dotenv';
```
Comment (john-whiting) [inline discord_bot/src/config.ts:5]: We should be able to get rid of this now since it is inside the devcontainer.

Comment (john-whiting) [reply discord_bot/src/config.ts:5]: Additionally, anything that uses this.

File: discord_bot/src/commands/verify.ts (hunk 9 above)
Comment (john-whiting) [inline discord_bot/src/commands/verify.ts:49]: It doesn't look like we are updating anything here when they are verified. Should we be doing this?

File: discord_bot/src/commands/verify.ts [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+              registrationYear: 2023,
+            },
+          });
+
+          console.log(participant)
+
+          if (!participant) {
+            reply = `I could not find a registration with the email: ${email}. Please make sure that the email you entered is correct!`;
+            await interaction.reply({ content: reply });
+          } else if (participant.verified) {
```
Comment (john-whiting) [inline discord_bot/src/commands/verify.ts:43]: The "verified" property on participants corresponds to the user responding to the registration verification email, not if they have linked their discord. We will likely want a different property here.

**Review by john-whiting — CHANGES_REQUESTED** [review]
I'd prefer if we did not use yarn v3 with pnp and instead stuck to yarn v1. Can you change this to yarn v1 instead? (The .yarn folder, with all its zip files, should no longer exist after this).

File: package.json [hunk 12]
```diff
@@ -30,7 +30,12 @@
     "install:server": "cd server && yarn install"
   },
   "dependencies": {
+    "@discordjs/rest": "^2.0.1",
```
Comment (john-whiting) [inline package.json:33]: Why are these being added to the root package.json? These should only exist inside the discord_bot/package.json file.

File: discord_bot/src/commands/verify.ts (hunk 11 above)
Comment (CatLuong0106) [reply discord_bot/src/commands/verify.ts:43]: What do you want to change it to

File: package.json (hunk 12 above)
Comment (CatLuong0106) [reply package.json:33]: Should I remove that then?

File: discord_bot/src/commands/schedule.ts (hunk 2 above)
Comment (CatLuong0106) [reply discord_bot/src/commands/schedule.ts:11]: Not sure. Ask Jaran

**Review by john-whiting — APPROVED** [review]

**Pull request merged**